File: Java/opentraceability/src/main/java/opentraceability/models/events/SensorMetaData.java
package opentraceability.models.events;

import java.net.URI;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityExtensionAttributesAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

public class SensorMetaData {

    @OpenTraceabilityJsonAttribute(name="time")
    @OpenTraceabilityAttribute(ns = "", name = "@time")
    public OffsetDateTime timeStamp;

    @OpenTraceabilityJsonAttribute(name="deviceID")
    @OpenTraceabilityAttribute(ns = "", name = "@deviceID")
    public URI deviceID;

    @OpenTraceabilityJsonAttribute(name="deviceMetadata")
    @OpenTraceabilityAttribute(ns = "", name = "@deviceMetadata")
    public URI deviceMetaData;

    @OpenTraceabilityJsonAttribute(name="rawData")
    @OpenTraceabilityAttribute(ns = "", name = "@rawData")
    public URI rawData;

    @OpenTraceabilityJsonAttribute(name="startTime")
    @OpenTraceabilityAttribute(ns = "", name = "@startTime")
    public OffsetDateTime startTime;

    @OpenTraceabilityJsonAttribute(name="bizRules")
    @OpenTraceabilityAttribute(ns = "", name = "@bizRules")
    public URI bizRules;

    @OpenTraceabilityJsonAttribute(name="dataProcessingMethod")
    @OpenTraceabilityAttribute(ns = "", name = "@dataProcessingMethod")
    public URI dataProcessingMethod;

    @OpenTraceabilityJsonAttribute(name="endTime")
    @OpenTraceabilityAttribute(ns = "", name = "@endTime")
    public OffsetDateTime endTime;

    @OpenTraceabilityExtensionAttributesAttribute
    public List<IEventKDE> extensionAttributes = new ArrayList<>();
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/models/events/sensormetadata.kt
package opentraceability.models.events

import opentraceability.interfaces.IEventKDE
import opentraceability.utility.attributes.*
import java.util.*
import java.net.URI
import java.time.OffsetDateTime

class SensorMetaData {

    @OpenTraceabilityJsonAttribute("time")
    @OpenTraceabilityAttribute("","@time")
    var timeStamp: OffsetDateTime? = null

    @OpenTraceabilityJsonAttribute("deviceID")
    @OpenTraceabilityAttribute("","@deviceID")
    var deviceID: URI? = null

    @OpenTraceabilityJsonAttribute("deviceMetadata")
    @OpenTraceabilityAttribute("","@deviceMetadata")
    var deviceMetaData: URI? = null

    @OpenTraceabilityJsonAttribute("rawData")
    @OpenTraceabilityAttribute("","@rawData")
    var rawData: URI? = null

    @OpenTraceabilityJsonAttribute("startTime")
    @OpenTraceabilityAttribute("","@startTime")
    var startTime: OffsetDateTime? = null

    @OpenTraceabilityJsonAttribute("bizRules")
    @OpenTraceabilityAttribute("","@bizRules")
    var bizRules: URI? = null

    @OpenTraceabilityJsonAttribute("dataProcessingMethod")
    @OpenTraceabilityAttribute("","@dataProcessingMethod")
    var dataProcessingMethod: URI? = null

    @OpenTraceabilityJsonAttribute("endTime")
    @OpenTraceabilityAttribute("","@endTime")
    var endTime: OffsetDateTime? = null

    @OpenTraceabilityExtensionAttributesAttribute
    var extensionAttributes: MutableList<IEventKDE> = mutableListOf()
}
